site-src/geps/gep-1424.md

```diff
@@ -0,0 +1,384 @@
+
+## API
+
+```
+metadata:
+  name: foo_route
+  namespace: store
+spec:
+  parentRefs:
+  - kind: Service
+    name: foo
+  rules:
+    backendRefs:
+    - kind: Service
+      name: foo
+      weight: 90
+    - kind: Service
+      name: foo_v2
+      weight: 10
+```
+
+In the example above, routing rules have been configured to direct 90% of traffic for the `foo` `Service` to the default "backend" endpoints specified by the `foo` `Service` [`selector`](https://kubernetes.io/docs/concepts/services-networking/service/#defining-a-service) field, and 10% to the `foo_v2` `Service`. This is determined based on the `ClusterIP` (for `Service`) and `ClusterSetIP` (for `ServiceImport`) matching, and for "transparent proxy" mesh implementations would match all requests to `foo.svc.cluster.local` (or arbitrary custom suffix, as the hostname is not specified manually) from within the same namespace, all requests to `foo.store.svc.cluster.local` from other namespaces, and all requests to `foo.store.svc.clusterset.local` for multicluster services, within the scope of the service mesh.
+
+### Omitting `backendRefs`
+
+Implementations SHOULD support a terse syntax which allows omitting `backendRefs` to avoid unnecessary redundancy for simple configurations. If no `backendRefs` are specified, implementations should direct traffic to the default endpoints of the `parentRef` `Service`. If the `parentRef` does not have any endpoints (e.g. if it is a `Service` without selectors), implementations MUST return an HTTP 503 response code (see discussion in [#1210](https://github.com/kubernetes-sigs/gateway-api/pull/1210)).
+
+This syntax has limited utility currently, but could become more relevant if additional functionality beyond traffic splitting is added to the `HTTPRoute` resource. `HTTPRoutes` configured in this way MAY be used to manually enroll services into a mesh (which could trigger behavior like injecting a sidecar proxy) if they have not already been enrolled by some other mechanism (as proposed in [GEP-1291: Mesh Representation](https://docs.google.com/document/d/1oyA9uUH7pNNxxwy3WZGSWx-edHDBLrujcezr8q3el70/edit) or similar).
+
+```
+metadata:
+  name: foo_route
+  namespace: store
+spec:
+  parentRefs:
+  - kind: Service
+    name: foo
+  rules:
+    matches:
+    - path:
+        value: "/bar"
+```
+
+The example above would drop all incoming traffic for HTTP paths other than `/bar` to the `foo` `Service`, following the existing spec to return HTTP 404 response codes for unmatched requests, and HTTP 500 response codes for requests excluded due to an [`HTTPRouteFilter`](https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io/v1beta1.HTTPRouteFilter).
```

Comment:
This is super subtle - I think its correct but should be more explicit.

I think this is saying:

* All traffic implicitly works without routes (this is just how K8s works)
* When you create a route you are *replacing* that implicit logic - not adding to it
* Therefor you may be restricting traffic via a route

Note: restriction != authz

Reply:
@howardjohn Would this specify the behavior more clearly?
```suggestion
The example above would drop all incoming traffic for HTTP paths other than `/bar` to the `foo` `Service`, following the existing spec to return HTTP 404 response codes for unmatched requests, and HTTP 500 response codes for requests excluded due to an [`HTTPRouteFilter`](https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io/v1beta1.HTTPRouteFilter).

When no `HTTPRoute` resources or AuthZ configuration are defined, all traffic should implicitly work - this is just how Kubernetes functions. When you create an `HTTPRoute` targeting a service as a `parentRef` you are replacing that implicit logic - not adding to it. Therefore, you may be reshaping or restricting traffic via an `HTTPRoute` configuration (which should be noted is distinct from disallowing traffic by AuthZ).
```

Reply:
:+1: